baybe/searchspace/core.py

```diff
@@ -146,6 +146,50 @@ def from_product(
 
         return SearchSpace(discrete=discrete, continuous=continuous)
 
+    @classmethod
+    def from_dataframe(
+        cls,
+        dataframe: pd.DataFrame,
+        parameters: List[Parameter],
+    ) -> SearchSpace:
+        """Create a search space from a specified set of parameter configuration.
+
+        The way in which the contents of the columns are interpreted depends on the
+        types of the corresponding parameter objects provided. For details, see
+        :meth:`baybe.searchspace.discrete.SubspaceDiscrete.from_dataframe` and
+        :meth:`baybe.searchspace.continuous.SubspaceContinuous.from_dataframe`.
+
+        Args:
+            dataframe: A dataframe whose parameter configurations are used as
+                search space specification.
+            parameters: The corresponding parameter objects, one for each column
+                in `dataframe`.
+
+        Returns:
+            The created search space.
+
+        Raises:
+            ValueError: If the dataframe columns do not match with the parameters.
+        """
+        if set(p.name for p in parameters) != set(dataframe.columns.values):
+            raise ValueError(
+                "The provided dataframe columns must match exactly with the specified "
+                "parameter names."
+            )
+
+        disc_params = [p for p in parameters if p.is_discrete]
+        cont_params = [p for p in parameters if not p.is_discrete]
+
+        # TODO: The continuous classmethod does not yet accept a parameter argument.
```

Comment:
thats a small change, I suggest to either include it or remove this todo (why do conti params need this argument?)

Reply:
Was mostly there to raise your attention, which seems to have worked 😄 I just noticed that slight inconsistency in the signatures of both methods. For discrete parameters, we certainly need the explicit parameter specification (e.g. to be able to fully configure them, explicitly choose their types, etc.). For conti parameters, this is not strictly needed at the moment because there is only one type available. However, while writing, I just noticed that we'd also need the argument if we want to explicitly set bounds for some of the parameters. So I guess I'll just add it, what do you think?

Reply:
yep agree

Reply:
Done. Have also extracted a utility to avoid code duplication for both subspaces. Along the way, have fixed the types and also rewritten the docstrings. Most importantly, have added tests for all three constructors to ensure they do what they are supposed to do. Hope that all open issues are resolved now. Can you have a look at the last few commits and confirm?

Reply:
gucci